## Pandas - изучение данных

Перейти по ссылке [Kaggle]( https://www.kaggle.com/openfoodfacts/world-food-facts)

Скачать датасет на свой компьютер и распаковать

Считать csv файл в датафрейм food

Посмотреть первые 5 записей

Сколько наблюдений в наборе данных?


In [16]:
import pandas as pd
a = pd.read_table('en.openfoodfacts.org.products.tsv')
a.head()


code                                                url  \
0   3087  http://world-en.openfoodfacts.org/product/0000...   
1   4530  http://world-en.openfoodfacts.org/product/0000...   
2   4559  http://world-en.openfoodfacts.org/product/0000...   
3  16087  http://world-en.openfoodfacts.org/product/0000...   
4  16094  http://world-en.openfoodfacts.org/product/0000...   

                      creator   created_t      created_datetime  \
0  openfoodfacts-contributors  1474103866  2016-09-17T09:17:46Z   
1             usda-ndb-import  1489069957  2017-03-09T14:32:37Z   
2             usda-ndb-import  1489069957  2017-03-09T14:32:37Z   
3             usda-ndb-import  1489055731  2017-03-09T10:35:31Z   
4             usda-ndb-import  1489055653  2017-03-09T10:34:13Z   

  last_modified_t last_modified_datetime                    product_name  \
0      1474103893   2016-09-17T09:18:13Z              Farine de blé noir   
1      1489069957   2017-03-09T14:32:37Z  Banana Chips Sweetened (Whole)   
2      1489069957   2017-03-09T14:32:37Z                         Peanuts   
3      1489055731   2017-03-09T10:35:31Z          Organic Salted Nut Mix   
4      1489055653   2017-03-09T10:34:13Z                 Organic Polenta   

  generic_name quantity         ...         fruits-vegetables-nuts_100g  \
0          NaN      1kg         ...                                 NaN   
1          NaN      NaN         ...                                 NaN   
2          NaN      NaN         ...                                 NaN   
3          NaN      NaN         ...                                 NaN   
4          NaN      NaN         ...                                 NaN   

  fruits-vegetables-nuts-estimate_100g collagen-meat-protein-ratio_100g  \
0                                  NaN                              NaN   
1                                  NaN                              NaN   
2                                  NaN                              NaN   
3                                  NaN                              NaN   
4                                  NaN                              NaN   

  cocoa_100g chlorophyl_100g carbon-footprint_100g nutrition-score-fr_100g  \
0        NaN             NaN                   NaN                     NaN   
1        NaN             NaN                   NaN                    14.0   
2        NaN             NaN                   NaN                     0.0   
3        NaN             NaN                   NaN                    12.0   
4        NaN             NaN                   NaN                     NaN   

  nutrition-score-uk_100g glycemic-index_100g water-hardness_100g  
0                     NaN                 NaN                 NaN  
1                    14.0                 NaN                 NaN  
2                     0.0                 NaN                 NaN  
3                    12.0                 NaN                 NaN  
4                     NaN                 NaN                 NaN  

[5 rows x 163 columns]

In [6]:
a.shape

(356027, 163)

Сколько столбцов в наборе данных?

In [8]:
len(a.columns)

163

Напечатайте все названия столбцов

In [9]:
a.columns

Index(['code', 'url', 'creator', 'created_t', 'created_datetime',
       'last_modified_t', 'last_modified_datetime', 'product_name',
       'generic_name', 'quantity',
       ...
       'fruits-vegetables-nuts_100g', 'fruits-vegetables-nuts-estimate_100g',
       'collagen-meat-protein-ratio_100g', 'cocoa_100g', 'chlorophyl_100g',
       'carbon-footprint_100g', 'nutrition-score-fr_100g',
       'nutrition-score-uk_100g', 'glycemic-index_100g',
       'water-hardness_100g'],
      dtype='object', length=163)

Какое название у 105 столбца?

In [10]:
a.columns[104]

'-glucose_100g'

Какой тип наблюдений в 105 столбце?

In [11]:
a.iloc[:, 104].dtype

dtype('float64')

Как проиндексирован набор данных?

In [14]:
a.index

RangeIndex(start=0, stop=356027, step=1)

Каково значение имени продукта в 19-м наблюдении?

In [13]:
a.loc[18,'product_name']

'Lotus Organic Brown Jasmine Rice'

## Pandas - фильтрация данных

Считать данные по  [адресу](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv). 
Присвоить переменной chipo

In [79]:
chipo = pd.read_table('https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv')
chipo

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
5            3         1                           Chicken Bowl   
6            3         1                          Side of Chips   
7            4         1                          Steak Burrito   
8            4         1                       Steak Soft Tacos   
9            5         1                          Steak Burrito   
10           5         1                    Chips and Guacamole   
11           6         1                   Chicken Crispy Tacos   
12           6         1                     Chicken Soft Tacos   
13           7         1                           Chicken Bowl   
14           7         1                    Chips and Guacamole   
15           8         1  Chips and Tomatillo-Green Chili Salsa   
16           8         1                        Chicken Burrito   
17           9         1                        Chicken Burrito   
18           9         2                            Canned Soda   
19          10         1                           Chicken Bowl   
20          10         1                    Chips and Guacamole   
21          11         1                       Barbacoa Burrito   
22          11         1                       Nantucket Nectar   
23          12         1                        Chicken Burrito   
24          12         1                                   Izze   
25          13         1           Chips and Fresh Tomato Salsa   
26          13         1                           Chicken Bowl   
27          14         1                       Carnitas Burrito   
28          14         1                            Canned Soda   
29          15         1                        Chicken Burrito   
...        ...       ...                                    ...   
4592      1825         1                       Barbacoa Burrito   
4593      1825         1                          Carnitas Bowl   
4594      1825         1                          Barbacoa Bowl   
4595      1826         1                           Chicken Bowl   
4596      1826         1                    Chips and Guacamole   
4597      1826         1                      Canned Soft Drink   
4598      1826         1                          Bottled Water   
4599      1827         1                           Chicken Bowl   
4600      1827         1                    Chips and Guacamole   
4601      1827         1                      Canned Soft Drink   
4602      1827         1                       Barbacoa Burrito   
4603      1827         1                       Barbacoa Burrito   
4604      1828         1                           Chicken Bowl   
4605      1828         1                    Chips and Guacamole   
4606      1828         1                      Canned Soft Drink   
4607      1829         1                          Steak Burrito   
4608      1829         1                         Veggie Burrito   
4609      1829         1                      Canned Soft Drink   
4610      1830         1                          Steak Burrito   
4611      1830         1                         Veggie Burrito   
4612      1831         1                          Carnitas Bowl   
4613      1831         1                                  Chips   
4614      1831         1                          Bottled Water   
4615      1832         1                     Chicken Soft Tacos   
4616      1832         1                    Chips and Guacamole   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken 

Как много продуктов стоят более чем $10.00?

In [80]:
chipo['price_modified'] = chipo['item_price'].apply(lambda x: x.replace('$', ''))
chipo['price_modified'] = chipo['price_modified'].apply(lambda x: x.replace(' ', ''))
chipo['price_modified'] = pd.to_numeric(chipo['price_modified'])
chipo

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
5            3         1                           Chicken Bowl   
6            3         1                          Side of Chips   
7            4         1                          Steak Burrito   
8            4         1                       Steak Soft Tacos   
9            5         1                          Steak Burrito   
10           5         1                    Chips and Guacamole   
11           6         1                   Chicken Crispy Tacos   
12           6         1                     Chicken Soft Tacos   
13           7         1                           Chicken Bowl   
14           7         1                    Chips and Guacamole   
15           8         1  Chips and Tomatillo-Green Chili Salsa   
16           8         1                        Chicken Burrito   
17           9         1                        Chicken Burrito   
18           9         2                            Canned Soda   
19          10         1                           Chicken Bowl   
20          10         1                    Chips and Guacamole   
21          11         1                       Barbacoa Burrito   
22          11         1                       Nantucket Nectar   
23          12         1                        Chicken Burrito   
24          12         1                                   Izze   
25          13         1           Chips and Fresh Tomato Salsa   
26          13         1                           Chicken Bowl   
27          14         1                       Carnitas Burrito   
28          14         1                            Canned Soda   
29          15         1                        Chicken Burrito   
...        ...       ...                                    ...   
4592      1825         1                       Barbacoa Burrito   
4593      1825         1                          Carnitas Bowl   
4594      1825         1                          Barbacoa Bowl   
4595      1826         1                           Chicken Bowl   
4596      1826         1                    Chips and Guacamole   
4597      1826         1                      Canned Soft Drink   
4598      1826         1                          Bottled Water   
4599      1827         1                           Chicken Bowl   
4600      1827         1                    Chips and Guacamole   
4601      1827         1                      Canned Soft Drink   
4602      1827         1                       Barbacoa Burrito   
4603      1827         1                       Barbacoa Burrito   
4604      1828         1                           Chicken Bowl   
4605      1828         1                    Chips and Guacamole   
4606      1828         1                      Canned Soft Drink   
4607      1829         1                          Steak Burrito   
4608      1829         1                         Veggie Burrito   
4609      1829         1                      Canned Soft Drink   
4610      1830         1                          Steak Burrito   
4611      1830         1                         Veggie Burrito   
4612      1831         1                          Carnitas Bowl   
4613      1831         1                                  Chips   
4614      1831         1                          Bottled Water   
4615      1832         1                     Chicken Soft Tacos   
4616      1832         1                    Chips and Guacamole   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken 

In [81]:
chipo['choice_description'] = chipo['choice_description'].fillna('Empty')
chipo['item_item'] = chipo['item_name'].map(str) +' with ' + chipo['choice_description']
chipo

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
5            3         1                           Chicken Bowl   
6            3         1                          Side of Chips   
7            4         1                          Steak Burrito   
8            4         1                       Steak Soft Tacos   
9            5         1                          Steak Burrito   
10           5         1                    Chips and Guacamole   
11           6         1                   Chicken Crispy Tacos   
12           6         1                     Chicken Soft Tacos   
13           7         1                           Chicken Bowl   
14           7         1                    Chips and Guacamole   
15           8         1  Chips and Tomatillo-Green Chili Salsa   
16           8         1                        Chicken Burrito   
17           9         1                        Chicken Burrito   
18           9         2                            Canned Soda   
19          10         1                           Chicken Bowl   
20          10         1                    Chips and Guacamole   
21          11         1                       Barbacoa Burrito   
22          11         1                       Nantucket Nectar   
23          12         1                        Chicken Burrito   
24          12         1                                   Izze   
25          13         1           Chips and Fresh Tomato Salsa   
26          13         1                           Chicken Bowl   
27          14         1                       Carnitas Burrito   
28          14         1                            Canned Soda   
29          15         1                        Chicken Burrito   
...        ...       ...                                    ...   
4592      1825         1                       Barbacoa Burrito   
4593      1825         1                          Carnitas Bowl   
4594      1825         1                          Barbacoa Bowl   
4595      1826         1                           Chicken Bowl   
4596      1826         1                    Chips and Guacamole   
4597      1826         1                      Canned Soft Drink   
4598      1826         1                          Bottled Water   
4599      1827         1                           Chicken Bowl   
4600      1827         1                    Chips and Guacamole   
4601      1827         1                      Canned Soft Drink   
4602      1827         1                       Barbacoa Burrito   
4603      1827         1                       Barbacoa Burrito   
4604      1828         1                           Chicken Bowl   
4605      1828         1                    Chips and Guacamole   
4606      1828         1                      Canned Soft Drink   
4607      1829         1                          Steak Burrito   
4608      1829         1                         Veggie Burrito   
4609      1829         1                      Canned Soft Drink   
4610      1830         1                          Steak Burrito   
4611      1830         1                         Veggie Burrito   
4612      1831         1                          Carnitas Bowl   
4613      1831         1                                  Chips   
4614      1831         1                          Bottled Water   
4615      1832         1                     Chicken Soft Tacos   
4616      1832         1                    Chips and Guacamole   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken 

In [82]:
chipo['corrected_price'] = chipo['price_modified'] / chipo['quantity']
chipo

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
5            3         1                           Chicken Bowl   
6            3         1                          Side of Chips   
7            4         1                          Steak Burrito   
8            4         1                       Steak Soft Tacos   
9            5         1                          Steak Burrito   
10           5         1                    Chips and Guacamole   
11           6         1                   Chicken Crispy Tacos   
12           6         1                     Chicken Soft Tacos   
13           7         1                           Chicken Bowl   
14           7         1                    Chips and Guacamole   
15           8         1  Chips and Tomatillo-Green Chili Salsa   
16           8         1                        Chicken Burrito   
17           9         1                        Chicken Burrito   
18           9         2                            Canned Soda   
19          10         1                           Chicken Bowl   
20          10         1                    Chips and Guacamole   
21          11         1                       Barbacoa Burrito   
22          11         1                       Nantucket Nectar   
23          12         1                        Chicken Burrito   
24          12         1                                   Izze   
25          13         1           Chips and Fresh Tomato Salsa   
26          13         1                           Chicken Bowl   
27          14         1                       Carnitas Burrito   
28          14         1                            Canned Soda   
29          15         1                        Chicken Burrito   
...        ...       ...                                    ...   
4592      1825         1                       Barbacoa Burrito   
4593      1825         1                          Carnitas Bowl   
4594      1825         1                          Barbacoa Bowl   
4595      1826         1                           Chicken Bowl   
4596      1826         1                    Chips and Guacamole   
4597      1826         1                      Canned Soft Drink   
4598      1826         1                          Bottled Water   
4599      1827         1                           Chicken Bowl   
4600      1827         1                    Chips and Guacamole   
4601      1827         1                      Canned Soft Drink   
4602      1827         1                       Barbacoa Burrito   
4603      1827         1                       Barbacoa Burrito   
4604      1828         1                           Chicken Bowl   
4605      1828         1                    Chips and Guacamole   
4606      1828         1                      Canned Soft Drink   
4607      1829         1                          Steak Burrito   
4608      1829         1                         Veggie Burrito   
4609      1829         1                      Canned Soft Drink   
4610      1830         1                          Steak Burrito   
4611      1830         1                         Veggie Burrito   
4612      1831         1                          Carnitas Bowl   
4613      1831         1                                  Chips   
4614      1831         1                          Bottled Water   
4615      1832         1                     Chicken Soft Tacos   
4616      1832         1                    Chips and Guacamole   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken 

In [83]:
corrected_price = chipo.loc[chipo['corrected_price'] > 10]
corrected_price['item_item'].count()

1034

# Как много продуктов стоят более чем $10.00? - ОТВЕТ: 1034

Какова цена каждого предмета? Напечатать столбцы item_name и item_price

In [96]:
b = chipo[['item_item','corrected_price']]
b.drop_duplicates(inplace = True)
b

c:\users\iv.andreev\appdata\local\programs\python\python37\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


item_item  corrected_price
0               Chips and Fresh Tomato Salsa with Empty             2.39
1                                Izze with [Clementine]             3.39
2                         Nantucket Nectar with [Apple]             3.39
3      Chips and Tomatillo-Green Chili Salsa with Empty             2.39
4     Chicken Bowl with [Tomatillo-Red Chili Salsa (...             8.49
5     Chicken Bowl with [Fresh Tomato Salsa (Mild), ...            10.98
6                              Side of Chips with Empty             1.69
7     Steak Burrito with [Tomatillo Red Chili Salsa,...            11.75
8     Steak Soft Tacos with [Tomatillo Green Chili S...             9.25
9     Steak Burrito with [Fresh Tomato Salsa, [Rice,...             9.25
10                       Chips and Guacamole with Empty             4.45
11    Chicken Crispy Tacos with [Roasted Chili Corn ...             8.75
12    Chicken Soft Tacos with [Roasted Chili Corn Sa...             8.75
13    Chicken Bowl with [Fresh Tomato Salsa, [Fajita...            11.25
16    Chicken Burrito with [Tomatillo-Green Chili Sa...             8.49
17    Chicken Burrito with [Fresh Tomato Salsa (Mild...             8.49
18                            Canned Soda with [Sprite]             1.09
19    Chicken Bowl with [Tomatillo Red Chili Salsa, ...             8.75
21    Barbacoa Burrito with [[Fresh Tomato Salsa (Mi...             8.99
22           Nantucket Nectar with [Pomegranate Cherry]             3.39
23    Chicken Burrito with [[Tomatillo-Green Chili S...            10.98
24                               Izze with [Grapefruit]             3.39
26    Chicken Bowl with [Roasted Chili Corn Salsa (M...             8.49
27    Carnitas Burrito with [[Tomatillo-Green Chili ...             8.99
28                        Canned Soda with [Dr. Pepper]             1.09
29    Chicken Burrito with [Tomatillo-Green Chili Sa...             8.49
31    Steak Burrito with [[Roasted Chili Corn Salsa ...             8.99
33    Carnitas Bowl with [Tomatillo-Red Chili Salsa ...             8.99
34                             Bottled Water with Empty             1.09
35    Chicken Soft Tacos with [Roasted Chili Corn Sa...             8.75
...                                                 ...              ...
4502  Chicken Soft Tacos with [Fresh Tomato Salsa, [...            11.25
4504  Carnitas Bowl with [Fresh Tomato Salsa, [Rice,...            11.75
4508                    Steak Burrito with [Brown Rice]             8.99
4509                                   Chips with Empty             1.99
4510                     Barbacoa Bowl with [Guacamole]            11.49
4518  Steak Bowl with [Tomatillo Red Chili Salsa, [R...            11.75
4530  Chicken Burrito with [Roasted Chili Corn Salsa...             8.75
4536  Steak Burrito with [Fresh Tomato Salsa, [Fajit...             9.25
4540  Chicken Bowl with [Tomatillo Green Chili Salsa...            11.25
4541  Veggie Salad Bowl with [Tomatillo Green Chili ...             8.75
4547  Steak Salad Bowl with [Roasted Chili Corn Sals...             9.39
4549  Steak Soft Tacos with [Tomatillo Green Chili S...             9.25
4553  Chicken Bowl with [Roasted Chili Corn Salsa, [...            11.25
4556  Chicken Burrito with [Tomatillo Green Chili Sa...            11.25
4557  Carnitas Soft Tacos with [Tomatillo Red Chili ...             9.25
4559  Chicken Burrito with [Tomatillo Red Chili Sals...            11.25
4562  Carnitas Bowl with [Roasted Chili Corn Salsa, ...             9.25
4566  Chicken Burrito with [Roasted Chili Corn Salsa...             8.75
4573  Veggie Salad Bowl with [Fresh Tomato Salsa, [F...             8.75
4589  Chicken Bowl with [Fresh Tomato Salsa, [Rice, ...            11.25
4592  Barbacoa Burrito with [Tomatillo Red Chili Sal...            11.75
4593  Carnitas Bowl with [Roasted Chili Corn Salsa, ...            11.75
4594  Barbacoa Bowl with [Roasted Chili Corn Salsa, ...            11.75
4595  Chicken Bowl with [Tomati

Отсортировать по имени предмета

In [100]:
b.sort_values('item_item')

item_item  corrected_price
357                       6 Pack Soft Drink with [Coke]             6.49
341                  6 Pack Soft Drink with [Diet Coke]             6.49
3141                  6 Pack Soft Drink with [Lemonade]             6.49
721                     6 Pack Soft Drink with [Nestea]             6.49
298                     6 Pack Soft Drink with [Sprite]             6.49
3195  Barbacoa Bowl with [Fresh Tomato (Mild), [Lett...             8.69
3822  Barbacoa Bowl with [Fresh Tomato (Mild), [Lett...             8.69
1175  Barbacoa Bowl with [Fresh Tomato Salsa (Mild),...            11.48
1804  Barbacoa Bowl with [Fresh Tomato Salsa (Mild),...             8.99
3017  Barbacoa Bowl with [Fresh Tomato Salsa, [Fajit...            11.75
418   Barbacoa Bowl with [Fresh Tomato Salsa, [Fajit...             9.25
4485  Barbacoa Bowl with [Fresh Tomato Salsa, [Fajit...            11.75
95    Barbacoa Bowl with [Fresh Tomato Salsa, [Fajit...             9.25
1347  Barbacoa Bowl with [Fresh Tomato Salsa, [Rice,...             9.25
2749  Barbacoa Bowl with [Fresh Tomato Salsa, [Rice,...            11.75
1643  Barbacoa Bowl with [Fresh Tomato Salsa, [Rice,...            11.75
3549  Barbacoa Bowl with [Fresh Tomato Salsa, [Rice,...             9.25
3376  Barbacoa Bowl with [Fresh Tomato Salsa, [Rice,...             9.25
1357  Barbacoa Bowl with [Fresh Tomato Salsa, [Rice,...             9.25
2620  Barbacoa Bowl with [Fresh Tomato Salsa, [Rice,...             9.25
3547  Barbacoa Bowl with [Fresh Tomato Salsa, [Rice,...            11.75
2253  Barbacoa Bowl with [Fresh Tomato Salsa, [Rice,...             9.25
4510                     Barbacoa Bowl with [Guacamole]            11.49
2399  Barbacoa Bowl with [Roasted Chili Corn Salsa, ...             9.25
39    Barbacoa Bowl with [Roasted Chili Corn Salsa, ...            11.75
3025  Barbacoa Bowl with [Roasted Chili Corn Salsa, ...            11.75
4594  Barbacoa Bowl with [Roasted Chili Corn Salsa, ...            11.75
2517  Barbacoa Bowl with [Roasted Chili Corn Salsa, ...             9.25
592   Barbacoa Bowl with [Roasted Chili Corn Salsa, ...             9.25
1419      Barbacoa Bowl with [Roasted Chili Corn Salsa]             9.25
...                                                 ...              ...
2482  Veggie Burrito with [[Tomatillo-Green Chili Sa...            10.98
2755  Veggie Burrito with [[Tomatillo-Green Chili Sa...             8.49
1653  Veggie Crispy Tacos with [Fresh Tomato Salsa (...             8.49
4201  Veggie Salad Bowl with [Fresh Tomato Salsa, [F...            11.25
295   Veggie Salad Bowl with [Fresh Tomato Salsa, [F...            11.25
960   Veggie Salad Bowl with [Fresh Tomato Salsa, [F...             8.75
4573  Veggie Salad Bowl with [Fresh Tomato Salsa, [F...             8.75
455   Veggie Salad Bowl with [Fresh Tomato Salsa, [F...            11.25
186   Veggie Salad Bowl with [Fresh Tomato Salsa, [F...            11.25
1316  Veggie Salad Bowl with [Fresh Tomato Salsa, [F...             8.75
4261  Veggie Salad Bowl with [Fresh Tomato Salsa, [F...            11.25
2269  Veggie Salad Bowl with [Fresh Tomato Salsa, [F...             8.75
3293  Veggie Salad Bowl with [Fresh Tomato Salsa, [R...             8.75
496   Veggie Salad Bowl with [Fresh Tomato Salsa, [R...            11.25
2223  Veggie Salad Bowl with [Roasted Chili Corn Sal...             8.75
2683  Veggie Salad Bowl with [Roasted Chili Corn Sal...             8.75
4541  Veggie Salad Bowl with [Tomatillo Green Chili ...             8.75
4109  Veggie Salad Bowl with [Tomatillo Red Chili Sa...            11.25
2156  Veggie Salad Bowl with [Tomatillo Red Chili Sa...            11.25
2996  Veggie Salad with [Roasted Chili Corn Salsa (M...             8.49
3163  Veggie Salad with [[Fresh Tomato Salsa (Mild),...             8.49
1694  Veggie Salad with [[Fresh Tomato Salsa (Mild),...             8.49
2756  Veggie Salad with [[Tomatillo-Green Chili Sals...             8.49
3889  Veggie Soft Tacos with [F

Сколько было заказано самого дорогого предмета?

In [106]:
max_price = max(b['corrected_price'])

In [109]:
b.loc[b['corrected_price'] == max_price].count()

item_item          28
corrected_price    28
dtype: int64

Сколько раз был заказан предмет "Veggie Salad Bowl"?

In [113]:
vsb = chipo.loc[chipo['item_name'] == "Veggie Salad Bowl"]
vsb.count()

order_id              18
quantity              18
item_name             18
choice_description    18
item_price            18
price_modified        18
item_item             18
corrected_price       18
a                     18
dtype: int64

# 18 раз

Сколько раз люди заказали более чем 1 предмет "Canned Soda"?

In [114]:
can_soda = chipo.loc[(chipo['item_name'] == 'Canned Soda') & (chipo['quantity'] > 1)]
can_soda.count()

order_id              20
quantity              20
item_name             20
choice_description    20
item_price            20
price_modified        20
item_item             20
corrected_price       20
a                     20
dtype: int64

# 20 раз